core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -1859,7 +1983,7 @@ private static boolean safeDouble(double d) {
     // negated MIN and MAX double values, overflow has not occurred. Otherwise,
     // overflow has occurred. Important to note that 'Double.MIN_VALUE' refers to
     // minimum positive value.
-    if (d < Double.MAX_VALUE && d > Double.MIN_VALUE) {
+    if (d < Double.MAX_VALUE && d > Double.MIN_VALUE || d == 0) {
```

Comment:
Why we have to check `d == 0` here?

Reply:
The way I check if a `Double` is "safe" or not is by checking whether it falls within the following ranges:

`Largest Negative Value` < `value` < `Smallest negative value` 

or

`Smallest positive value` < `value` < `Largest positive value`

The exception to this is zero, so I check for that as well.

Let me know if this makes sense or you think it should be changed.

Reply:
Get it. Then how about adjusting it like this: 
```
return (d < Double.MAX_VALUE && d > Double.MIN_VALUE)
    || (d > -Double.MAX_VALUE && d < -Double.MIN_VALUE)
    || d == 0;
```

Reply:
DOne